### PR TITLE
Fix bug adding certificate to linux vmss, flatten exception and fix service fabric durability upgrade issue

### PR DESCRIPTION
**Pull request by a-santamaria**
## Description

- Fixing add certificate to Linux vmss. Set CertificateStore to null as this is an invalid parameter for Linux.

- Fixing Azure/service-fabric-issues#1054 flatten inner exception (KeyVaultClient.GetSecretAsync) so it is displayed correctly.

- Fixing Update Service Fabric durability. Waiting for cluster configuration update to finish before starting vmss CreateOrUpdate operation, if the configuration upgrade fails there is no need to send the vmss operation, just show the exception.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCertificateCmdlet.cs [hunk 1]
```diff
@@ -586,7 +594,16 @@ private string GetThumbprintFromSecret(string secretUrl)
...
 
-            var secretBundle = this.KeyVaultClient.GetSecretAsync(secretUrl).Result;
+            SecretBundle secretBundle;
+            try
+            {
+                secretBundle = this.KeyVaultClient.GetSecretAsync(secretUrl).Result;
+            }
+            catch (Exception ex)
+            {
+                throw GetInnerException(ex);
```
Comment (linmeng08) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCertificateCmdlet.cs]: Any particular reason for this? This usage replaces the callstack of original exception and should be avoided unless there's a strong reason.

Comment (a-santamaria) [reply src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCertificateCmdlet.cs]: Not doing this was the cause of this issue: Azure/service-fabric-issues#1054 where people had to see the exception with $error[0].Exception.InnerExceptions

**Comment by MiYanni** [comment]
@azuresdkci Retest this please

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/UpdateAzureRmServiceFabricDurability.cs [hunk 2]
```diff
@@ -125,11 +125,10 @@ public override void ExecuteCmdlet()
                     NodeTypes = cluster.NodeTypes
                 };
 
-                var patchTask = PatchAsync(patchArg);
+                var psCluster = SendPatchRequest(patchArg);
 
-                WriteClusterAndVmssVerboseWhenUpdate(new List<Task>() { vmssTask, patchTask }, true, this.NodeType);
```
Comment (juhacket) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/UpdateAzureRmServiceFabricDurability.cs]: Why is this change needed? The VMSS and SF updates should be able to run in parallel.

**Comment by MiYanni** [comment]
@azuresdkci Retest this please
